llama_stack/providers/utils/kvstore/redis/redis.py

```diff
@@ -48,5 +48,27 @@ async def delete(self, key: str) -> None:
     async def range(self, start_key: str, end_key: str) -> List[str]:
         start_key = self._namespaced_key(start_key)
         end_key = self._namespaced_key(end_key)
+        cursor = 0
+        pattern = start_key + "*"  # Match all keys starting with start_key prefix
+        matching_keys = []
+        while True:
+            cursor, keys = await self.redis.scan(cursor, match=pattern, count=1000)
 
-        return await self.redis.zrangebylex(start_key, end_key)
+            for key in keys:
+                key_str = key.decode("utf-8") if isinstance(key, bytes) else key
+                if start_key <= key_str <= end_key:
```

Comment:
can we use a more strict pattern like `pattern = f"{namespace}[{start_key}-{end_key}]"` to only fetch the keys in range instead of having to filter again here?